File: src/main/java/cn/xunyard/idea/coding/log/window/WindowLogger.java
package cn.xunyard.idea.coding.log.window;

import cn.xunyard.idea.coding.log.Logger;
import cn.xunyard.idea.coding.util.ProjectUtils;
import lombok.RequiredArgsConstructor;

/**
 * @author <a herf="mailto:wuqi@terminus.io">xunyard</a>
 * @date 2019-12-26
 */
@RequiredArgsConstructor
public class WindowLogger extends AbstractThreadBasedWindowedLogger implements Logger {

    private final String identity;

    private LogMessage buildLogMessage(LogOperation operation, String format, Object... args) {
        return new LogMessage(ProjectUtils.getCurrentProject(), identity, operation, format, args);
    }

    @Override
    public void clear() {
        submitLog(buildLogMessage(LogOperation.CLEAR, null));
    }

    @Override
    public void debug(String format, Object... args) {
        submitLog(buildLogMessage(LogOperation.DEBUG, format, args));
    }

    @Override
    public void info(String format, Object... args) {
        submitLog(buildLogMessage(LogOperation.INFO, format, args));
    }

    @Override
    public void warn(String format, Object... args) {
        submitLog(buildLogMessage(LogOperation.WARN, format, args));
    }

    @Override
    public void error(String format, Object... args) {
        submitLog(buildLogMessage(LogOperation.ERROR, format, args));
    }

    @Override
    public void done() {
        submitLog(buildLogMessage(LogOperation.DONE, null));
    }
}

File: src/main/kotlin/cn/xunyard/idea/coding/log/window/WindowLogger.kt
package cn.xunyard.idea.coding.log.window

import cn.xunyard.idea.coding.log.Logger
import cn.xunyard.idea.coding.util.ProjectUtils.getCurrentProject

/**
 * @author <a herf="mailto:xunyard@gmail.com">xunyard</a>
 * @date 2019-12-26
 */
class WindowLogger constructor(
        private val identity: String
) : AbstractThreadBasedWindowedLogger(), Logger {

    private fun buildLogMessage(operation: LogOperation, format: String?, vararg args: Any): LogMessage {
        return LogMessage(getCurrentProject(), identity, operation, format, args)
    }

    override fun clear() {
        submitLog(buildLogMessage(LogOperation.CLEAR, null))
    }

    override fun debug(format: String, vararg args: Any) {
        submitLog(buildLogMessage(LogOperation.DEBUG, format, *args))
    }

    override fun info(format: String, vararg args: Any) {
        submitLog(buildLogMessage(LogOperation.INFO, format, *args))
    }

    override fun warn(format: String, vararg args: Any) {
        submitLog(buildLogMessage(LogOperation.WARN, format, *args))
    }

    override fun error(format: String, vararg args: Any) {
        submitLog(buildLogMessage(LogOperation.ERROR, format, *args))
    }

    override fun done() {
        submitLog(buildLogMessage(LogOperation.DONE, null))
    }
}